### PR TITLE
Set default language for TNTSearch

### PR DESCRIPTION
**Pull request by Koc**
### 📝 Summary
This change improves morphology in search functionality by providing correct Stemmer for TNTSearch

#### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
image  | image |

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [ ] Tests (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [ ] Documentation ([README](https://github.com/nextcloud/collectives/blob/main/README.md) or [documentation](https://github.com/nextcloud/collectives/blob/main/docs/)) has been updated or is not required

### CODE REVIEW

**Comment by Koc** [comment]
@mejo- @juliushaertl can you review?

File: lib/Search/FileSearch/FileSearcher.php [hunk 1]
```diff
@@ -21,13 +21,29 @@ class FileSearcher extends TNTSearch {
...
+		'pt' => 'Portuguese',
+		'ru' => 'Russian',
+		'uk' => 'Ukrainian',
+	];
+
 	protected FileIndexer $indexer;
+	private string $language;
 
-	public function __construct() {
+	public function __construct(string $language = 'no') {
```
Comment (juliushaertl) [inline lib/Search/FileSearch/FileSearcher.php:41]:
```suggestion
	public function __construct(?string $language = null) {
```

Would prefer this to not be confused with no for Norwegian

**Review by juliushaertl — COMMENTED** [review]
Looks good to me with one comment regarding the default value

File: lib/Search/FileSearch/FileSearcher.php (hunk 1 above)
Comment (Koc) [reply lib/Search/FileSearch/FileSearcher.php]: fixed, but we should to bypass `'No'` explicitly to `\TeamTNT\TNTSearch\Indexer\TNTIndexer::setLanguage`

Comment (Koc) [reply lib/Search/FileSearch/FileSearcher.php]: @juliushaertl can you re-check/approve? :pray:

**Review by juliushaertl — APPROVED** [review]

File: lib/Search/FileSearch/FileSearcher.php [hunk 2]
```diff
@@ -21,13 +21,31 @@ class FileSearcher extends TNTSearch {
...
+		'en' => 'Porter',
+		'it' => 'Italian',
+		'lv' => 'Latvian',
+		'pl' => 'Polish',
+		'pt' => 'Portuguese',
+		'ru' => 'Russian',
+		'uk' => 'Ukrainian',
+	];
+
+	private const UNSUPPORTED_LANGUAGE = 'No';
```
Comment (mejo-) [inline lib/Search/FileSearch/FileSearcher.php:38]: Shouldn't this be some string that is less likely to be similar to `no` (for "Norwegian")?

Comment (mejo-) [reply lib/Search/FileSearch/FileSearcher.php:38]: Ok, thanks for the pointer. Then the PR is good to go :blush: Thanks for the contribution @Koc!

**Pull request merged**